main.py

```diff
@@ -49,63 +48,86 @@ def main(input_path, output_path):
+        if total_file_count == 0:
+            logging.error("No files found at your input location")
+            return
+
+        
+
+        # Iterate over all images in the input directory
+        # and get text from each image
+        other_files = 0
+        successful_files = 0
+        logging.info("Found total {} file(s)\n".format(total_file_count))
+        for ctr, filename in enumerate(os.listdir(input_path)):
+            logging.debug("Parsing {}".format(filename))
+            extension = os.path.splitext(filename)[1]
+
+            if extension.lower() not in VALID_IMAGE_EXTENSIONS:
+                other_files += 1
+                continue
+
+            image_file_name = os.path.join(input_path, filename)
+            filename_without_extension = os.path.splitext(filename)[0]
+            text_file_path = os.path.join(output_path, filename_without_extension)
+            subprocess.run(['tesseract', image_file_name, text_file_path],
+                           stdout=subprocess.PIPE,
+                           stderr=subprocess.PIPE)
+            logging.debug("Successfully parsed {}".format(filename))
+            successful_files += 1
+
+        logging.info("Parsing Completed!\n")
+        if successful_files == 0:
+            logging.error("No valid image file found.")
+            logging.error("Supported formats: [{}]".format(", ".join(VALID_IMAGE_EXTENSIONS)))
+        else:
+            logging.info("Successfully parsed images: {}".format(successful_files))
+            logging.info("Files with unsupported file extensions: {}".format(other_files))
+    else:
+        filename = os.path.basename(input_path)
         filename_without_extension = os.path.splitext(filename)[0]
         text_file_path = os.path.join(output_path, filename_without_extension)
-        subprocess.run(['tesseract', image_file_name, text_file_path],
-                       stdout=subprocess.PIPE,
-                       stderr=subprocess.PIPE)
-        logging.debug("Successfully parsed {}".format(filename))
-        successful_files += 1
-
-    logging.info("Parsing Completed!\n")
-    if successful_files == 0:
-        logging.error("No valid image file found.")
-        logging.error("Supported formats: [{}]".format(", ".join(VALID_IMAGE_EXTENSIONS)))
-    else:
-        logging.info("Successfully parsed images: {}".format(successful_files))
-        logging.info("Files with unsupported file extensions: {}".format(other_files))
+        subprocess.run(['tesseract', filename, text_file_path],
+                           stdout=subprocess.PIPE,
+                           stderr=subprocess.PIPE)
 
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser()
-    parser.add_argument('--input_dir', help="Input directory where input images are stored", required=True)
+    parser.add_argument('--input_dir', help="Input directory where input images are stored")
+    parser.add_argument('--file', help="Input image")
```

Comment:
You can use `input_file` as the parameter name
```suggestion
    parser.add_argument('--input_file', help="Input image filepath")
```